js/activity.js

```diff
@@ -3429,22 +3424,108 @@ class Activity {
                     }
                 }
             }
-
+            activity.textMsg(
+                _("Item restored from the trash."),
+                3000 
+            );
+        
             this.refreshCanvas();
-        };
+        }; 
+
+        // Add event listener for trash icon click
+        document.getElementById('restoreIcon').addEventListener('click', () => {
+            this._renderTrashView();
+        }); 
+
+        // function to hide trashView from canvas
+        function handleClickOutsideTrashView(trashView) {
+            let firstClick = true;
+            document.addEventListener('click', (event) => {
+                if (firstClick) {
+                    firstClick = false;
+                    return;
+                }
+                if (!trashView.contains(event.target) && event.target !== trashView) {
+                    trashView.style.display = 'none';
+                }
+            });
+        }
 
-        this.handleKeyDown = (event) => {
-            
-            if (event.ctrlKey && event.key === "z") {
-                this._restoreTrash(activity);
-                activity.__tick();
-                event.preventDefault();
+        this._renderTrashView = () => {
+            if (!activity.blocks || !activity.blocks.trashStacks || activity.blocks.trashStacks.length === 0) {
+                return;
+            }
+            const trashList = document.getElementById('trashList');
+            const trashView = document.createElement('div');
+            trashView.id = 'trashView';
+            trashView.classList.add('trash-view');
+        
+            // Sticky buttons
+            const buttonContainer = document.createElement('div');
+            buttonContainer.classList.add('button-container');
+        
+            const restoreLastBtn = document.createElement('button');
+            restoreLastBtn.textContent = 'Restore Last';
```

Comment:
This string needs i18n. _("Restore Last") 

Same for Restore All on Line 3477.

Also, we should use "Restore last" and "Restore all" to be consistent with the other menu labels.